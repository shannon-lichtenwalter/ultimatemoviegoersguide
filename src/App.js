import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import Nav from './Components/Nav/Nav';
import HomePage from './Components/HomePage/HomePage';
import SearchPage from './Components/SearchPage/SearchPage';
import AboutPage from './Components/AboutPage/AboutPage';
import MovieDetails from './Components/MovieDetails/MovieDetails';
import ErrorPage from './Components/ErrorPage/ErrorPage';
import './index.css';

class App extends React.Component {
  state = {
    error: null,
    movieDetails: null
  }

  setError = (e) => {
    this.setState({
      error: e
    })
    this.props.history.push('/error')
  }

  resetError = () => {
    this.setState({
      error: null
    })
  }

  showDetails = (data) => {
    this.setState({
      movieDetails: data
    }, this.props.history.push('/movie-details'))
  }


  render() {
    return (
      <main className='App'>
        <Nav />
        <header>
          <h1>Ultimate Moviegoers Guide</h1>
        </header>
        <Switch>
          <Route
            exact
            path={'/'}
            component={() => <HomePage
              error={this.state.error}
              setError={this.setError}
              resetError={this.resetError}
              showDetails={this.showDetails}
            />}
          />

          <Route
            exact
            path={'/search'}
            component={() => <SearchPage
              error={this.state.error}
              resetError={this.resetError}
              showDetails={this.showDetails}
              />}
          />

          <Route
            exact
            path={'/about'}
            component={() => <AboutPage
              error={this.state.error}
              resetError={this.resetError}
              />}
          />

          <Route
            exact
            path={'/movie-details'}
            component={() => <MovieDetails
              movieDetails= {this.state.movieDetails}
              error={this.state.error}
              resetError={this.resetError}
              />}
          />

          <Route
            exact
            path={'/error'}
            component={ErrorPage}
          />
        </Switch>

      </main>
    );
  }
}

export default withRouter(App);
