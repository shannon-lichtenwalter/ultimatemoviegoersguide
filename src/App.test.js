import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

describe('App Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render
      (<BrowserRouter><App /></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  }); 

  it('Displays the App component correctly when rendered', () => {
    const wrapper = mount(<BrowserRouter><App /></BrowserRouter>);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});
