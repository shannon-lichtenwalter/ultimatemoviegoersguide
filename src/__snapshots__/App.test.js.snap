// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Displays the App component correctly when rendered 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(App)>
      <App
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <main
          className="App"
        >
          <withRouter(Nav)>
            <Nav
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <nav>
                <button
                  onClick={[Function]}
                >
                  Home
                </button>
                <button
                  onClick={[Function]}
                >
                  Search
                </button>
                <button
                  onClick={[Function]}
                >
                  About
                </button>
              </nav>
            </Nav>
          </withRouter(Nav)>
          <header>
            <h1>
              Ultimate Moviegoers Guide
            </h1>
          </header>
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            >
              <component
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <HomePage
                  error={null}
                  resetError={[Function]}
                  setError={[Function]}
                  showDetails={[Function]}
                >
                  <section>
                    <form>
                      <label
                        hidden={true}
                        htmlFor="listSelect"
                      >
                        Displaying:
                      </label>
                      <select
                        id="listSelect"
                        onChange={[Function]}
                      >
                        <option
                          value="now_playing"
                        >
                          Now Playing
                        </option>
                        <option
                          value="popular"
                        >
                          Popular
                        </option>
                        <option
                          value="top_rated"
                        >
                          Top Rated
                        </option>
                      </select>
                    </form>
                    <ul
                      className="movies"
                    />
                  </section>
                </HomePage>
              </component>
            </Route>
          </Switch>
        </main>
      </App>
    </withRouter(App)>
  </Router>
</BrowserRouter>
`;
